src/pkg/build/build.go

```diff
@@ -281,17 +268,21 @@ func (b *Build) insertMetadata() (err error) {
 
 func (b *Build) runPreScript() error {
 	if b.runPre() && b.d.BuildData.Pre != "" {
-		// Run %pre script here
-		pre := exec.Command("/bin/sh", "-cex", b.d.BuildData.Pre)
-		pre.Stdout = os.Stdout
-		pre.Stderr = os.Stderr
-
-		sylog.Infof("Running pre scriptlet\n")
-		if err := pre.Start(); err != nil {
-			sylog.Fatalf("failed to start %%pre proc: %v\n", err)
-		}
-		if err := pre.Wait(); err != nil {
-			sylog.Fatalf("pre proc: %v\n", err)
+		if syscall.Getuid() == 0 {
+			// Run %pre script here
+			pre := exec.Command("/bin/sh", "-cex", b.d.BuildData.Pre)
+			pre.Stdout = os.Stdout
+			pre.Stderr = os.Stderr
+
+			sylog.Infof("Running pre scriptlet\n")
+			if err := pre.Start(); err != nil {
+				sylog.Fatalf("failed to start %%pre proc: %v\n", err)
+			}
+			if err := pre.Wait(); err != nil {
+				sylog.Fatalf("pre proc: %v\n", err)
+			}
+		} else {
+			sylog.Errorf("Attempted to build with scripts as non-root user")
```

Comment:
Not actually returning an error, just writing an error message then returning nil!!